### PR TITLE
Prometheus: (Dataplane) Set FrameTypeVersion on Scalar type responses

### PR DESCRIPTION
**Pull request by kylebrandt**
**What is this feature?**

Set Dataplane FrameTypeVersion on Scalar type responses

**Why do we need this feature?**

So expressions (SSE) and recorded queries (RQ) detect scalar responses response correctly (e.g. 1+1).

### CODE REVIEW

File: pkg/util/converter/prom.go [hunk 1]
```diff
@@ -662,7 +666,7 @@ func readMatrixOrVectorMulti(iter *jsonitere.Iterator, resultType string, opt Op
 				Type:   data.FrameTypeTimeSeriesMulti,
 				Custom: resultTypeToCustomMeta(resultType),
 			}
-			if opt.Dataplane && resultType == "vector" {
+			if opt.Dataplane && resultType == "vector" || resultType == "scalar" {
```
Comment (toddtreece) [inline pkg/util/converter/prom.go:669]: should the OR be wrapped in parentheses?
```suggestion
			if opt.Dataplane && (resultType == "vector" || resultType == "scalar") {
```

File: pkg/util/converter/prom.go [hunk 2]
```diff
@@ -662,7 +666,7 @@ func readMatrixOrVectorMulti(iter *jsonitere.Iterator, resultType string, opt Op
 				Type:   data.FrameTypeTimeSeriesMulti,
 				Custom: resultTypeToCustomMeta(resultType),
 			}
-			if opt.Dataplane && resultType == "vector" {
+			if opt.Dataplane && (resultType == "vector" || resultType == "scalar") {
```
Comment (toddtreece) [inline pkg/util/converter/prom.go:669]: looking at this again, and i'm wondering if this is needed? do/should scalar results hit this function?

Comment (kylebrandt) [reply pkg/util/converter/prom.go:669]: It does not.

**Review by toddtreece — APPROVED** [review]

File: pkg/util/converter/prom.go [hunk 3]
```diff
@@ -564,6 +564,10 @@ func readScalar(iter *jsonitere.Iterator) backend.DataResponse {
 		Custom: resultTypeToCustomMeta("scalar"),
 	}
 
+	if dataPlane {
+		frame.Meta.TypeVersion = data.FrameTypeVersion{0, 1}
+	}
```
Comment (samjewell) [inline pkg/util/converter/prom.go:569]: @kylebrandt could/should we unit test this? Seems we've added a conditional to our application code, but no test to specify the behaviour.

**Pull request merged**